torchbenchmark/models/timm_nfnet/__init__.py

```diff
@@ -1,87 +1,115 @@
+        else:
+            _logger.info('Testing with a single process on 1 GPU.')
+        if not device or device == "cpu":
+            args.prefetcher = False
+        else:
+            args.prefetcher = not args.no_prefetcher
+            assert args.prefetcher, "Test requires the data to be prefetched during execution"
+        # resolve AMP arguments based on PyTorch amp availability
+        args.use_amp = None
+        if args.amp and has_native_amp():
+            args.use_amp = 'native'
+        args.model_name = variant
+        args.batch_size = train_bs
+        args.eval_batch_size = eval_bs
+        args.train_num_batch = TRAIN_NUM_BATCH
+        args.eval_num_batch = EVAL_NUM_BATCH
+        self.args = args
 
-    def _gen_target(self, batch_size):
-        return torch.empty(
-            (batch_size,) + self.cfg.target_shape,
-            device=self.device, dtype=torch.long).random_(self.cfg.num_classes)
-
-    def _step_train(self):
-        self.cfg.optimizer.zero_grad()
-        output = self.model(self.cfg.example_inputs)
-        if isinstance(output, tuple):
-            output = output[0]
-        target = self._gen_target(output.shape[0])
-        self.cfg.loss(output, target).backward()
-        self.cfg.optimizer.step()
-
-    # vision models have another model
-    # instance for inference that has
-    # already been optimized for inference
-    def set_eval(self):
-        pass
-
-    def _step_eval(self):
-        self.eval_model(self.cfg.infer_example_inputs)
+        model, self.loader_train, self.loader_validate, self.optimizer, \
+            self.train_loss_fn, self.lr_scheduler, self.amp_autocast, \
+            self.loss_scaler, self.mixup_fn, self.validate_loss_fn = timm_instantiate_train(args)
+        eval_model, self.loader_eval = timm_instantiate_eval(args)
+        # jit the model if required
+        self.model, self.eval_model = jit_if_needed(model, eval_model, jit=jit)
+        
+        # Disable train data load as there is not enough GPU memory
+        # TODO: enable with larger GPU
+        # self.loader_train = prefetch_loader(self.loader_train, device)
+        # self.loader_validate = prefetch_loader(self.loader_validate, device)
+        self.loader_eval = prefetch_loader(self.loader_eval, device)
 
     def get_module(self):
-        return self.model, (self.cfg.example_inputs,)
+        raise NotImplementedError("Disable the train test because it causes CUDA OOM on Nvidia T4")
+        self.eval_model.eval()
+        with torch.no_grad():
+            for _, (input, _) in zip(range(self.args.eval_num_batch), self.loader_eval):
+                return self.eval_model, (input, )
 
+    # Temporarily disable training because this will cause CUDA OOM in CI
+    # TODO: re-enable this test when better hardware is available
     def train(self, niter=1):
+        raise NotImplementedError("Disable the train test because it causes CUDA OOM on Nvidia T4")
```

Comment:
Would this also disable the test for CPU?

Reply:
Thanks for the comment! I have enabled it for CPU.